org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/contentmodel/XSDDocumentation.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.xsd.contentmodel;
+
+import org.apache.xerces.xs.XSObjectList;
+import org.eclipse.lemminx.settings.XSDDocumentationSource;
+import org.eclipse.lemminx.utils.StringUtils;
+
+public class XSDDocumentation {
+	private String prefix;
+	private String documentation;
+	private String appinfo;
+
+	public XSDDocumentation(XSObjectList annotations, XSDDocumentationSource docStrategy) {
+		this(annotations, null, docStrategy);
+	}
+
+	public XSDDocumentation(XSObjectList annotations, String value, XSDDocumentationSource docStrategy) {
+		String documentation = "";
+		String appinfo = "";
+		switch(docStrategy) {
+			case both: {
+				documentation = XSDAnnotationModel.getDocumentation(annotations, value);
+				appinfo = XSDAnnotationModel.getAppInfo(annotations, value);
+				break;
+			}
+			case documentation: {
+				documentation = XSDAnnotationModel.getDocumentation(annotations, value);
+				break;
+			}
+			case appinfo: {
+				appinfo = XSDAnnotationModel.getAppInfo(annotations, value);
+				break;
+			}
+		}
+		this.documentation = documentation;
+		this.appinfo = appinfo;
+		this.prefix = XSDAnnotationModel.getPrefix(annotations, value);
+	}
+
+	public String getFormattedDocumentation(boolean html) {
+		StringBuilder doc = new StringBuilder();
+		boolean bothExists = !StringUtils.isEmpty(documentation) && !StringUtils.isEmpty(appinfo);
+
+		if (!StringUtils.isEmpty(documentation)) {
+			if (html) {
+				doc.append("<p>");
+			}
+			if (bothExists) {
+				doc.append(applyPrefix("documentation", html));
+			}
+			doc.append(this.getDocumentation());
+			if (html) {
+				doc.append("</p>");
+			}
+		}
+
+		if (!StringUtils.isEmpty(appinfo)) {
+			if (html) {
+				doc.append("<p>");
+			}
+			if (bothExists) {
+				doc.append(applyPrefix("appinfo", html));
+			}
+			doc.append(this.getAppinfo());
+			if (html) {
+				doc.append("</p>");
+			}
+		}
+		return doc.toString();
+	}
+
+	private String applyPrefix(String documentation, boolean html) {
+		StringBuilder result = new StringBuilder();
```

Comment:
I wonder if xs:documentation, xs:appinfo should be embed inside a p element to have a cariage return after the xs documentation. @fbricon what do you think about that?

Reply:
In my latest push, there is a new line after xs:documentation and xs:appinfo

Reply:
great thanks.